pkg/apis/networking/v1/types.go

```diff
@@ -174,3 +175,144 @@ type NetworkList struct {
 
 	Items []Network `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// RemoteCluster is a specification for a RemoteCluster resource
+type RemoteCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RemoteClusterSpec   `json:"spec"`
+	Status RemoteClusterStatus `json:"status"`
+}
+
+// RemoteClusterSpec is the spec for a RemoteCluster resource
+type RemoteClusterSpec struct {
+	ConnConfig APIServerConnConfig `json:"connConfig"`
+}
+
+type APIServerConnConfig struct {
+	// apiServer address. Format: https://ip:port
+	Endpoint   string `json:"endpoint"`
+	CABundle   []byte `json:"caBundle"`
+	ClientCert []byte `json:"clientCert"`
+	ClientKey  []byte `json:"clientKey"`
+	// The maximum length of time to wait before giving up on a server
+	// request. A value of zero means no timeout. Default: zero second
+	Timeout uint32 `json:"timeout"`
+}
+
+// RemoteClusterStatus is the status for a RemoteCluster resource
+type RemoteClusterStatus struct {
+	// Conditions is an array of current cluster conditions.
+	Conditions []ClusterCondition `json:"conditions"`
+}
+
+// ClusterCondition describes current state of a cluster.
+type ClusterCondition struct {
+	// Type of cluster condition, Ready or Offline.
+	Type ClusterConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status apiv1.ConditionStatus `json:"status"`
+	// Last time the condition was checked.
+	LastProbeTime metav1.Time `json:"lastProbeTime"`
+	// Last time the condition transit from one status to another.
+	// +optional
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+	// (brief) reason for the condition's last transition.
+	// +optional
+	Reason *string `json:"reason,omitempty"`
+	// Human readable message indicating details about last transition.
+	// +optional
+	Message *string `json:"message,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RemoteClusterList is a list of RemoteCluster resources
+type RemoteClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []RemoteCluster `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// RemoteSubnet is a specification for a RemoteSubnet resource
+type RemoteSubnet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RemoteSubnetSpec   `json:"spec"`
+	Status RemoteSubnetStatus `json:"status"`
+}
+
+// RemoteSubnetSpec is the spec for a RemoteSubnet resource
+type RemoteSubnetSpec struct {
+	Version      IPVersion   `json:"version"`
+	CIDR         string      `json:"cidr"`
```

Comment:
Cannot just use a CIDR string to describe a rama ip address range, rama supports a discontinuous ip address range description. (excludeIPs, start, end)

Reply:
In my opinion, ip address range description is not needed. Remote subnet and subnet are one-to-many relationship，and we only need the cidr for 40000 and 10000+ route table, right? @SwimilTylers

Reply:
excludeIPs won't be recorded in "podiplist", reserved ip too. @hhyasdf

Reply:
I guess it is caused by latest pr of `main`. We shall discuss about it @Louhwz .

Reply:
> In my opinion, ip address range description is not needed. Remote subnet and subnet are one-to-many relationship，and we only need the cidr for 40000 and 10000+ route table, right? @SwimilTylers

For local overlay subnet to access remote underlay subnet, if an ip address is excluded in remote underlay subnet, this ip address should be handled as an external ip, which is not in both local and remote cluster. So we need the information of excluded ip (and 'start', 'end').

Reply:
> excludeIPs won't be recorded in "podiplist", reserved ip too. @hhyasdf

Information of "podiplist" is used to assist the process of kernel vxlan neigh resolving, without it, kernel vxlan neigh resolving will also work out by policy routes. So accurate policy routes is needed.

Reply:
I got the meaning of "excluded ip", but why do we need 'start' and 'end'?

Reply:
> I got the meaning of "excluded ip", but why do we need 'start' and 'end'? @hhyasdf

We need them all. IP addresses beyond the range of 'start' and 'end' are also excluded ip addresses.

Reply:
> I got the meaning of "excluded ip", but why do we need 'start' and 'end'? @hhyasdf

'excluedIPs' is not the same with 'excluded ip addresses'.

Reply:
> > I got the meaning of "excluded ip", but why do we need 'start' and 'end'? @hhyasdf
> 
> 'excludeIPs' is not the same with 'excluded ip addresses'.

yeah, i got this.

Reply:
ok, i see. 3q